pkg/kube/controllers/extendedstatefulset/reconciler.go

```diff
@@ -258,3 +283,61 @@ func (r *ReconcileExtendedStatefulSet) getActualStatefulSet(ctx context.Context,
 
 	return result, maxVersion, nil
 }
+
+// listStatefulSetVersions gets all StatefulSets' versions and ready status owned by the ExtendedStatefulSet
+func (r *ReconcileExtendedStatefulSet) listStatefulSetVersions(ctx context.Context, exStatefulSet *essv1a1.ExtendedStatefulSet) (map[int]bool, error) {
+	result := map[int]bool{}
+
+	statefulSets, err := r.listStatefulSets(ctx, exStatefulSet)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, statefulSet := range statefulSets {
+		strVersion, ok := statefulSet.Annotations[essv1a1.AnnotationVersion]
+		if !ok {
+			strVersion = "0"
```

Comment:
I am not sure this is a correct assumption.
Can you describe a case where this would happen?
I'm thinking this should be an error :/

Reply:
Return error directly. I thought it again and statefulSet should have this annotation otherwise it had some inconsistent issue.